src/coins.cpp

```diff
@@ -253,11 +253,13 @@ bool CCoinsViewCache::Flush(bool will_reuse_cache) {
     auto cursor{CoinsViewCacheCursor(m_sentinel, cacheCoins, /*will_erase=*/true)};
     bool fOk = base->BatchWrite(cursor, hashBlock);
     if (fOk) {
-        cacheCoins.clear();
         if (will_reuse_cache) {
```

Comment:
we can likely get rid of the `will_reuse_cache` now that we have a reusable cache that we can reset